### PR TITLE
Add funcionality for Channel PoI with accounts and contracts

### PR DESCRIPTION
**Pull request by velzevur**
PT: [SC WS: Get PoI](https://www.pivotaltracker.com/story/show/159406178)

### CODE REVIEW

**Comment by velzevur** [comment]
Will add a corresponding description in the protocol.

File: apps/aehttp/src/sc_ws_handler.erl [hunk 1]
```diff
@@ -329,6 +329,38 @@ process_incoming(#{<<"action">> := <<"get">>,
...
+                {ok, PoI} ->
+                    Resp = #{action => <<"get">>,
+                             tag => <<"poi">>,
+                             payload => #{
+                                <<"poi">> => aec_base58c:encode(poi, aec_trees:serialize_poi(PoI))}},
+                    {reply, Resp};
+                {error, Reason} ->
+                    {reply, error_response(R, Reason)}
+            end;
+        _ -> {reply, error_response(R, broken_hexcode)}
```
Comment (lucafavatella) [inline apps/aehttp/src/sc_ws_handler.erl]: I suspect user would prefer a more specific error - i.e. on which of the two values the error is. Though seems good enough.

**Comment by velzevur** [comment]
While improving tests, found an issue.

**Comment by velzevur** [comment]
Improved error message for broken encoding to be more descriptive
Improved behaviour that keus `accounts` and `contracts` are no longer mandatorary in the payload
Fixed a small bug
Improved tests with negative paths

File: apps/aehttp/src/sc_ws_handler.erl [hunk 2]
```diff
@@ -329,6 +329,45 @@ process_incoming(#{<<"action">> := <<"get">>,
...
+                              [Res | Acc]
+                      end, [], Keys)}
+            catch
+                error:_ ->
+                    {error, invalid_pubkey}
+            end
+        end,
+    BrokenEncodingReply =
+        fun(What) ->
+            {reply, error_response(R, <<"broken_encoding: ", What/binary>>)}
```
Comment (uwiger) [inline apps/aehttp/src/sc_ws_handler.erl]: I don't think the error responses from failed `safe_decode` are entirely consistent throughout this API. Perhaps something to track in a PT?

File: apps/aehttp/test/aehttp_integration_SUITE.erl [hunk 3]
```diff
@@ -4764,9 +4764,57 @@ sc_ws_contract_(Config, TestName, Owner) ->
...
+                            }),
+
+                    {ok, #{<<"reason">> := R}} = ?WS:wait_for_channel_event(ConnPid, error),
+                    R
+                end,
+
+    EncodedPoI = GetPoI(SenderConnPid),
+    EncodedPoI = GetPoI(AckConnPid),
+
+    NegativePoiTests =
```
Comment (uwiger) [inline apps/aehttp/test/aehttp_integration_SUITE.erl]: Perhaps these should be broken out into a separate test? Could also be done some other time.

File: apps/aehttp/src/sc_ws_handler.erl (hunk 2 above)
Comment (velzevur) [reply apps/aehttp/src/sc_ws_handler.erl]: Tracked [SC WS: revisit broken encoding responses](https://www.pivotaltracker.com/story/show/159459679)

File: apps/aehttp/test/aehttp_integration_SUITE.erl (hunk 3 above)
Comment (velzevur) [reply apps/aehttp/test/aehttp_integration_SUITE.erl]: Tracked [SC WS Tests: revisit tests](https://www.pivotaltracker.com/story/show/159459719)

**Pull request merged**